lib/matplotlib/testing/compare.py

```diff
@@ -154,15 +154,17 @@ def encode_and_escape(name):
             + b") run flush\n")
         self._proc.stdin.flush()
         # GS> if nothing left on the stack; GS<n> if n items left on the stack.
-        err = self._read_until(b"GS")
-        stack = self._read_until(b">")
-        if stack or not os.path.exists(dest):
-            stack_size = int(stack[1:]) if stack else 0
-            self._proc.stdin.write(b"pop\n" * stack_size)
-            # Using the systemencoding should at least get the filenames right.
-            raise ImageComparisonFailure(
-                (err + b"GS" + stack + b">")
-                .decode(sys.getfilesystemencoding(), "replace"))
+        err = self._read_until((b"GS<", b"GS>"))
```

Comment:
AFAICT `_read_until` doesn't support tuples as is (`len(terminator)` is wrong) and needs to be fixed to support them.

Reply:
You are correct, I was so happy about figuring out that `.endswith()` accepts tuples that I've missed it ;-).

Reply:
Ok, so I've figured out that if instead of passing an explicit tuple, I use `*terminator` arg, I can allow multiple terminators without having to actually change the call sites or having to add a compatibility hack. If you don't like that, I can revert to adding `if not isinstance(terminator, tuple)` or changing all the call sites.

Reply:
Heh, wrong. I've just realized that stripping actually makes the `err.endswith()` test wrong :-/.

Reply:
`*terminator` seems nicer, actually.

Reply:
Heh, actually no other method even uses the return value, so I guess I can change it.

Reply:
Ok, I think it's ready now. I've noticed another late-night mistake, and fixed it. I've aimed for minimal change / keeping the code simple now.

Reply:
I wonder if the whole thing would be simpler by having terminator be a regex instead, and return `match.groups()`?  This way you can just match `r"GS(?:<(\d+))>"` instead of having to do two read_untils.  (The slight extra cost of doing regex matches should be negligible compared to interacting with the subprocess.)

(If doing so, I would suggest having the argument be always a *compiled* regex object so that callers have to call `re.compile` themselves to make it completely clear that the argument is a regex.)

Reply:
To be honest, I don't really want to spend more time on it and I don't think it's really important how it's implemented given it's only used in tests.

Reply:
no worries

Reply:
Thanks.